src/components/calendar/Calendar.tsx

```diff
@@ -217,7 +224,25 @@ export default function Calendar() {
     const scheduledEvent = scheduledEvents.find((se) => se.id === id);
 
     if (scheduledEvent) {
-      setSelectedEvent(scheduledEvent);
+      if (
+        scheduledEvent.proposalBooking &&
+        scheduledEvent.bookingType === ScheduledEventBookingType.USER_OPERATIONS
+      ) {
+        // NOTE: Types here are a bit of a mess. It needs more attention to fix.
```

Comment:
I stumbled upon this problem as well that the Types are not easy to use, maybe we should create a task and take it into a sprint, otherwise, it will be forgotten and the more we build on this the more time it will take later to fix.

Reply:
I think I fixed some of the types in one of the coming PRs but we can create a task anyway to check better.